packages/mui-material/src/Chip/Chip.js

```diff
@@ -156,141 +158,190 @@ const ChipRoot = styled('div', {
+              : alpha(
+                  theme.palette.action.selected,
+                  theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
+                ),
+          },
+          '&:active': {
+            boxShadow: (theme.vars || theme).shadows[1],
+          },
         },
-      }),
-  }),
-  ({ theme, ownerState }) => ({
-    ...(ownerState.variant === 'outlined' && {
-      backgroundColor: 'transparent',
-      border: theme.vars
-        ? `1px solid ${theme.vars.palette.Chip.defaultBorder}`
-        : `1px solid ${
-            theme.palette.mode === 'light' ? theme.palette.grey[400] : theme.palette.grey[700]
-          }`,
-      [`&.${chipClasses.clickable}:hover`]: {
-        backgroundColor: (theme.vars || theme).palette.action.hover,
-      },
-      [`&.${chipClasses.focusVisible}`]: {
-        backgroundColor: (theme.vars || theme).palette.action.focus,
-      },
-      [`& .${chipClasses.avatar}`]: {
-        marginLeft: 4,
-      },
-      [`& .${chipClasses.avatarSmall}`]: {
-        marginLeft: 2,
-      },
-      [`& .${chipClasses.icon}`]: {
-        marginLeft: 4,
-      },
-      [`& .${chipClasses.iconSmall}`]: {
-        marginLeft: 2,
-      },
-      [`& .${chipClasses.deleteIcon}`]: {
-        marginRight: 5,
-      },
-      [`& .${chipClasses.deleteIconSmall}`]: {
-        marginRight: 3,
       },
-    }),
-    ...(ownerState.variant === 'outlined' &&
-      ownerState.color !== 'default' && {
-        color: (theme.vars || theme).palette[ownerState.color].main,
-        border: `1px solid ${
-          theme.vars
-            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / 0.7)`
-            : alpha(theme.palette[ownerState.color].main, 0.7)
-        }`,
-        [`&.${chipClasses.clickable}:hover`]: {
-          backgroundColor: theme.vars
-            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / ${
-                theme.vars.palette.action.hoverOpacity
-              })`
-            : alpha(theme.palette[ownerState.color].main, theme.palette.action.hoverOpacity),
-        },
-        [`&.${chipClasses.focusVisible}`]: {
-          backgroundColor: theme.vars
-            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / ${
-                theme.vars.palette.action.focusOpacity
-              })`
-            : alpha(theme.palette[ownerState.color].main, theme.palette.action.focusOpacity),
-        },
-        [`& .${chipClasses.deleteIcon}`]: {
-          color: theme.vars
-            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / 0.7)`
-            : alpha(theme.palette[ownerState.color].main, 0.7),
-          '&:hover, &:active': {
-            color: (theme.vars || theme).palette[ownerState.color].main,
+      ...Object.entries(theme.palette)
+        .filter(([, value]) => value.dark)
+        .map(([color]) => ({
+          props: { color },
+          style: {
+            [`&.${chipClasses.clickable}`]: {
```

Comment:
This bumps specificity, there might be a way to do it with https://github.com/mui/material-ui/tree/master/packages/pigment-css-react#styling-based-on-runtime-values. Should I try? Any other options?

It's important to note that `ownerState.clickable` !== `props.clickable`

Reply:
> It's important to note that ownerState.clickable !== props.clickable

hmmm, this sounds like a problem to me.

Reply:
Removed specificity problem as per https://github.com/mui/material-ui/pull/41592#discussion_r1536115576